### PR TITLE
inspect IOException immediate cause if there are no details in the exception

### PR DESCRIPTION
**Pull request by shivanesabharwal**

### CODE REVIEW

File: src/main/java/net/starschema/clouddb/jdbc/BQSQLException.java [hunk 1]
```diff
@@ -176,7 +176,10 @@ private static String enrichReason(String reason, Throwable cause) {
         return prefix + googleJsonResponseException.getDetails().getMessage();
       }
     } else {
-      return prefix + cause.toString();
+      String throwableDetails = cause.toString();
+      // if the `cause` itself does not have a detailed message, ask for the message of its cause.
+      if (!throwableDetails.isEmpty()) throwableDetails = cause.getCause().toString();
```
Comment (tjbanghart) [inline src/main/java/net/starschema/clouddb/jdbc/BQSQLException.java:181]: Could `getCause` be null here?

Comment (shivanesabharwal) [reply src/main/java/net/starschema/clouddb/jdbc/BQSQLException.java:181]: indeed, i'm fixing

File: src/main/java/net/starschema/clouddb/jdbc/BQSQLException.java [hunk 2]
```diff
@@ -176,7 +176,13 @@ private static String enrichReason(String reason, Throwable cause) {
         return prefix + googleJsonResponseException.getDetails().getMessage();
       }
     } else {
-      return prefix + cause.toString();
+      String throwableDetails = cause.toString();
+      // if the `cause` itself does not have a detailed message, ask for the message of its cause.
+      if (!throwableDetails.isEmpty()) {
+        Throwable parentCause = cause.getCause();
```
Comment (shivanesabharwal) [inline src/main/java/net/starschema/clouddb/jdbc/BQSQLException.java:182]: @tjbanghart we could also attempt to walk the cause chain until we find something non-null

Comment (tjbanghart) [reply src/main/java/net/starschema/clouddb/jdbc/BQSQLException.java:182]: Let's start with this and revisit if it doesn't help.

**Pull request merged**